sklearn/svm/src/liblinear/linear.cpp

```diff
@@ -70,6 +77,22 @@ static void info(const char *fmt,...)
 static void info(const char *fmt,...) {}
 #endif
 
+// New function to ensure the same behaviour for random number generation on windows and linux
+// note that we always use the same seed (12574 here, why not) to get reproducible results
+#if INT_MAX == 0x7FFFFFFF
+std::mt19937 mt_rand(12574);
```

Comment:
This no longer depends on the user-provided random state?

Reply:
Good catch. I was remembering seeing something about user-provided seed in the sources, thanks for pointing out! I'll have a look. Unfortunately I messed up my windows sklearn environment and am currently facing the far-too-familiar-but-yet-not-understood error

```bash
=================================== ERRORS ====================================
______________________________ ERROR collecting  ______________________________
C:\Miniconda3\envs\sklearn\lib\site-packages\_pytest\config\__init__.py:429: in _importconftest
    return self._conftestpath2mod[conftestpath]
E   KeyError: local('C:\\_dev\\python_ws\\_Libs_OpenSource\\scikit-learn\\sklearn\\externals\\conftest.py')

During handling of the above exception, another exception occurred:
sklearn\__check_build\__init__.py:44: in <module>
    from ._check_build import check_build  # noqa
E   ModuleNotFoundError: No module named 'sklearn.__check_build._check_build'
```

Reply:
I'll try to modify the parameters structure so that a seed can be passed. Indeed as opposed to `srand()/rand()` there does not seem to be a global way to pass an initial seed to `mt19937` objects.